sway/input/seat.c

```diff
@@ -1347,14 +1351,16 @@ void seat_set_focus_layer(struct sway_seat *seat,
 			seat_set_focus(seat, previous);
 		}
 		return;
-	} else if (!layer || seat->focused_layer == layer) {
+	} else if (!layer) {
 		return;
 	}
 	assert(layer->surface->mapped);
 	seat_set_focus_surface(seat, layer->surface, true);
 	if (layer->current.layer >= ZWLR_LAYER_SHELL_V1_LAYER_TOP) {
 		seat->focused_layer = layer;
 	}
+	seat_set_focus_surface(seat, layer->surface, true);
+	seat->focused_layer = layer;
```

Comment:
I think you forgot to add something that sets `has_exclusive_layer`, as it is currently never set to true. Maybe that is what you *meant* to write here?

The check that the layer is TOP or above should also be respected, so it would likely look something like:

```c
if (layer >= TOP) {
   focused_layer = layer;
   if (interactivity == EXCLUSIVE) {
     has_exclusivity = true;
   }
 }
 ```

Reply:
Oh, merged the wrong conflict hunk... Fixing :)